scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1624,56 +1671,95 @@
-                     psuedo-matches [{:task-metadata-seq task-metadata-seq}]
-                     task-txns (map (fn [job metadata] (job->task-txn job metadata compute-cluster)) jobs task-metadata-seq)]
-                 (launch-tasks-for-cluster compute-cluster psuedo-matches task-txns pool-name conn nil false)))]
+       (let [distributor-fn (partial scheduling-capacity-constrained-job-distributor 
+                                     compute-cluster->scheduling-capacity)
+             kubernetes-jobs-launcher (create-kubernetes-jobs-launcher conn pool-name mesos-run-as-user)]
          (log-structured/info "Launching jobs in Kubernetes" {:pool pool-name})
-         (distribute-and-launch-jobs considerable-jobs pool-name compute-clusters-for-pool
-                                     job->acceptable-compute-clusters-fn launch-distributed-job-fn)
-         (log-structured/info "Launched jobs in Kubernetes" {:pool pool-name :number-launched-jobs (count considerable-jobs)}))))))
+         (distribute-and-launch-jobs considerable-jobs pool-name compute-clusters
+                                     job->acceptable-compute-clusters-fn
+                                     distributor-fn
+                                     kubernetes-jobs-launcher))))))
 
 (defn handle-kubernetes-scheduler-pool
   "Handle scheduling pending jobs using the Kubernetes Scheduler."
   [conn pool-name->pending-jobs-atom pool-name scheduler-config compute-clusters
    job->acceptable-compute-clusters-fn user->quota user->usage-future mesos-run-as-user]
   (log-structured/info "Running handler for Kubernetes Scheduler pool" {:pool pool-name})
   (try
-    (let [user->usage (tracing/with-span [s {:name "scheduler.kubernetes-handler.resolve-user-to-usage-future"
-                                             :tags {:pool pool-name :component tracing-component-tag}}]
-                        @user->usage-future)
-          {max-considerable :max-jobs-considered} scheduler-config
-          db (db conn)
-          pending-jobs (get @pool-name->pending-jobs-atom pool-name)
-          considerable-jobs (pending-jobs->considerable-jobs db pending-jobs user->quota user->usage max-considerable pool-name)
-          considerable-job-uuids (set (map :job/uuid considerable-jobs))]
-      (log-structured/info "Considering jobs" {:pool pool-name :number-considered-jobs (count considerable-job-uuids)})
-      (if (seq considerable-jobs)
+    (let [; This is required to properly filter compute clusters to
+          ; those configured for this pool. This is done via the synthetic-pods
+          ; config, accessed via the autoscaling? method. In the future, this list
+          ; of pools should be slightly restructured to be listed one level higher,
+          ; on the compute-cluster-template config.
+          compute-clusters-for-pool (filter #(cc/autoscaling? % pool-name) compute-clusters) 
+          max-total-considerable (:max-jobs-considered scheduler-config)
```

Comment:
These three vars can be destructured out from `scheduler=config` at the function argument stage:
`{:keys [max-totalconsiderable ...] :as scheduler-config} `

Reply:
done